frontend/src/main/scala/bloop/data/Project.scala

```diff
@@ -199,4 +200,70 @@ object Project {
         }
     }
   }
+
+  def pprint(projects: Traversable[Project]): String = {
+    projects.map(p => s"'${p.name}'").mkString(", ")
+  }
+
+  /**
+   * Enable any Metals-specific setting in a project by applying an in-memory
+   * project transformation. A setting is Metals-specific if it's required for
+   * Metals to provide a complete IDE experience to users.
+   *
+   * A side-effect of this transformation is that we force the resolution of the
+   * semanticdb plugin. This is an expensive operation that is heavily cached
+   * inside [[bloop.engine.caches.SemanticDBCache]] and which can be retried in
+   * case the resolution for a version hasn't been successful yet and the
+   * workspace settings passed as a parameter asks for another attempt.
+   *
+   * @param project The project that we want to transform.
+   * @param settings The settings that contain Metals-specific information such
+   *                 as the expected semanticdb version or supported Scala versions.
+   * @param logger The logger responsible of tracking any transformation-related event.
+   * @return Either the same project as before or the transformed project.
+   */
+  def enableMetalsSettings(
+      project: Project,
+      semanticDBPlugin: Option[AbsolutePath],
+      workspaceDir: AbsolutePath,
+      logger: Logger
+  ): Project = {
+    def enableSemanticDB(options: List[String], pluginPath: AbsolutePath): List[String] = {
```

Comment:
Instead of adding semanticdb and range pos in one pass, we do it in two different methods so that by default we add range pos and only if we have a plugin path we enable the semanticdb.